pyccel/codegen/printing/ccode.py

```diff
@@ -227,14 +227,15 @@ class CCodePrinter(CodePrinter):
         'dereference': set()
     }
 
-    def __init__(self, parser, settings=None):
+    def __init__(self, parser, target_language, settings=None):
```

Comment:
Why do you need the target language? If we are using `CCodePrinter` then the `target_language` is C.

The only place where this may not be true is in the `CWrapperCodePrinter` but if something needs to be done differently in the `CWrapperCodePrinter` then this should be done by overriding the function from `CCodePrinter` not by modifying the `CCodePrinter` class. Think of the `CWrapperCodePrinter` as a special case of a `CCodePrinter`. And the `CCodePrinter` as a printer for writing pure c (nothing that uses python functions or targets other languages)

Reply:
yeah this is better will change it